### PR TITLE
Updating README with NEO 3.0 and 2.x directions and perspectives

### PR DESCRIPTION
**Pull request by vncoelho**

### CODE REVIEW

**Comment by erikzhang** [comment]
Some things have not yet been determined.

**Comment by canesin** [comment]
@vncoelho I think that https://github.com/neo-project/neo/issues/727 may be a better direction. Just create a new repository for NEO3 with all the basic components on it ( cli + core + plugins ). It will make much easier track development and versions.
This repository could be kept for NEO2 maintaining only, the README have a legacy note and point to NEO3 repository.

**Comment by vncoelho** [comment]
Ok, @canesin, I am going to talk to @igormcoelho this afternoon and we talk again later today.

Let's see if we create a new one or merge everything here.

**Comment by shargon** [comment]
I agree, if we have all in the same repository, separated by projects like now, but all connected, is easier to maintain, now we can't use neo-cli with the current neo 3 code

**Comment by vncoelho** [comment]
Yes, @erikzhang, that is why it was a draft, currently representing the current state.

However, maybe we should follow that direction of creating a new repository.
I do not see plenty of problems in having NEO2.x in branches.
However, it looks like that the majority is preferring a new repo. But I believe that a repository for NEO2 would be better than a fresh one for NEO3.

I think that the most important thing might be to unify `neo-cli`, `neo-plugins` and `neo`.

**Comment by vncoelho** [comment]
Drafts can be updated direct here on github! :+1: 

I will mark this as Ready for review.
I believe that this could be merged.
If we create another repo for 2.x we can just update the readme.

Please note that the branch for neo-plugins was not yet created.
Could you proceed with that, @erikzhang?

File: README.md [hunk 1]
```diff
@@ -14,13 +14,30 @@
...
+From 2017 until middle 2020, NEO 2.x will still be supported and compatibility of new packages will be mantained.
+
+A link to the essential C# reference implementation can be seen below:
+
+* [neo-core](https://github.com/neo-project/neo/tree/master-2.x)
+* [neo-vm](https://github.com/neo-project/neo-vm/tree/master-2.x)
+* [neo-cli](https://github.com/neo-project/neo-cli/tree/master-2.x)
+* [neo-plugins](https://github.com/neo-project/neo-plugins/tree/master-2.x)
+
+NEO 1.x was known as Antsharess, the roots of such codes can be found historically in this current repository.
```
Comment (shargon) [inline README.md]:
```suggestion
NEO 1.x was known as Antshares, the roots of such codes can be found historically in this current repository. 
```

**Comment by erikzhang** [comment]
The white paper is out of date.

**Comment by vncoelho** [comment]
White paper? ahauahah
I did not understand.

File: README.md [hunk 2]
```diff
@@ -14,13 +14,30 @@
 
 </p>
 
-NEO: A distributed network for the Smart Economy
+NEO 3.0 (under development): A distributed network for the Smart Economy
 ================
 
 NEO uses digital identity and blockchain technology to digitize assets and leverages smart contracts for autonomously managed digital assets to create a "smart economy" within a decentralized network.
 
 To learn more about NEO, please read the [White Paper](http://docs.neo.org/en-us/index.html)|[白皮书](http://docs.neo.org/zh-cn/index.html).
```
Comment (erikzhang) [inline README.md]: The white paper is out of date.

Comment (vncoelho) [reply README.md]: Remove the "Under development"

File: README.md [hunk 3]
```diff
@@ -14,12 +14,29 @@
...
 </p>
 
-NEO: A distributed network for the Smart Economy
+NEO 3.0 (under development): A distributed network for the Smart Economy
 ================
 
 NEO uses digital identity and blockchain technology to digitize assets and leverages smart contracts for autonomously managed digital assets to create a "smart economy" within a decentralized network.
 
-To learn more about NEO, please read the [White Paper](http://docs.neo.org/en-us/index.html)|[白皮书](http://docs.neo.org/zh-cn/index.html).
+To learn more about NEO, please read the [White Paper](https://docs.neo.org/en-us/whitepaper.html)|[白皮书](https://docs.neo.org/zh-cn/whitepaper.html).
```
Comment (shargon) [inline README.md]:
```suggestion
To learn more about NEO, please read the [White Paper](https://docs.neo.org/en-us/whitepaper.html)|[白皮书](https://docs.neo.org/en-us/whitepaper.htmll).
```

Comment (vncoelho) [reply README.md]: I think that this was not really a good suggestion...aehuaheuaheauea I will revert it.

Comment (shargon) [inline README.md]: Should be `en-us` if you show `en-us`

Comment (vncoelho) [reply README.md]: I did not understand. There are two links, one for en-us (White Paper) and one for zh-cn (白皮书)

Comment (shargon) [reply README.md]: I thought that is the same xD, sorry :P

Comment (vncoelho) [reply README.md]: jajajajaj

File: README.md [hunk 4]
```diff
@@ -14,12 +14,29 @@
...
+
+NEO 2.x: Reference template  
+================
+
+NEO adopts a model of continuous improvement and evolution.
+We believe that Blockchain field of research and development will be kept in continous transformation for the next years.
```
Comment (EdgeDLT) [inline README.md]: > We believe that the research and development of blockchain technology will be kept in continuous transformation for the next years.

File: README.md [hunk 5]
```diff
@@ -14,12 +14,29 @@
...
+From 2017 until middle 2020, NEO 2.x will still be supported and compatibility of new packages will be mantained.
```
Comment (EdgeDLT) [inline README.md]: > From 2017 until the middle of 2020 (estimated), NEO 2.x will still be supported and the compatibility of new packages will be maintained.

File: README.md [hunk 6]
```diff
@@ -14,12 +14,29 @@
...
+
+A link to the essential C# reference implementation can be seen below:
+
+* [neo-core](https://github.com/neo-project/neo/tree/master-2.x)
+* [neo-vm](https://github.com/neo-project/neo-vm/tree/master-2.x)
+* [neo-cli](https://github.com/neo-project/neo-cli/tree/master-2.x)
+* [neo-plugins](https://github.com/neo-project/neo-plugins/tree/master-2.x)
+* [neo-compiler](https://github.com/neo-project/neo-compiler/tree/master-2.x)
+
+NEO 1.x was known as Antshares, the roots of such codes can be found historically in this current repository.
```
Comment (EdgeDLT) [inline README.md]: > NEO 1.x was known as Antshares and the roots of the code can be found historically in this current repository.

**Comment by vncoelho** [comment]
You're the spokesperson @EdgeDLT, you said it's spoken.
Let's improve it. Thanks,

**Comment by erikzhang** [comment]
I have asked @dahongfei to modify this file. Please wait for him.

**Comment by vncoelho** [comment]
Such a honor, @dahongfei.
Fell free to reopen this PR and filter the parts you find important.

We tried to keep it simple and pass the message.

**Comment by vncoelho** [comment]
By the way, @dahongfei. Chapter "Towards the Smart Economy: the three pillars of NEO" is waiting for you: https://github.com/NeoResearch/specification-neo3.0/blob/master/sections/09_SmartEconomy.yaml

https://github.com/NeoResearch/specification-neo3.0/blob/master/sections/09_SmartEconomy.md

Aheuahueahuea
We keep in touch during these next months.

**Comment by vncoelho** [comment]
Noooo....ahauahahau
@erikzhang had just asked to wait...kkkkk

**Comment by vncoelho** [comment]
Let's open another one and update any other necessary info, @dahongfei.

**Comment by shargon** [comment]
sorry, we can update it in another pr?

**Comment by vncoelho** [comment]
No worries, @shargon. You mind if focused on several critical and outstanding things! Go for it.

I think that we will need another PR, because we also archived the the neo-compiler repo and other minor things that will probably come.

**Pull request merged**